### ISSUE TITLE
How to use RBAC or any Role and permission based  approach in nuxt?

### ISSUE THREAD

**Issue by sudhir600** [nuxt/nuxt.js#4676]
So basically i wanted to implement a secure dashboard and related pages. 
The things is different routes/actions should be accessible by either any roles or by even a user which has certain permission.

For example -` /user/add` can be accessible by any user who have access of user module but when route came `/user/update` then only own record can be see in list and can be updated. 
if same user access `/user/delete/{id}` then he can not even its own record. 

Some user can have delete own record and some have to delete other as well.

In short my application want permission on every single activity like this snapshot.
<br />
![image](https://user-images.githubusercontent.com/4952560/50625270-f262c080-0f4c-11e9-99d1-7f5904bd22f9.png)

**Comment by FreddyCrugger** [comment]
I got the same problem, did you manage to find a solution? I need to build a nuxt app with 2 roles but cannot work out how to add roles/permission via nuxt, no help online at all.

Thanks

**Comment by awronski** [comment]
In my app I have a frontend (nuxt) and the backend (server with the rest api)

1. User login to the app
2. Backend returns list of permission. I put them in the store
3. In nuxt I use `v-if` to show/hide options base on user permission.
4. Even if a user "hack" a frontend and enabled hidden options the backend check permission with every request. 
5. I user jwt-token to check credentials on the backend.

**Comment by FreddyCrugger** [comment]
Thanks.

I did find a way to do it later on.
I created 3 middlewares called:
auth-admin.js
auth-customers.js
auth-users.js

each file having this code:
export default function ({ store, redirect }) {
  // If the user is not authenticated
  if ((!store.state.isLoggedIn) && (store.state.rid !== "admin")) {  //or user or customer
    return redirect('/404');
  }
}


+ 1 called
auth.js

I then create 3 folders:
admin
customers
users

In the main layout I added at the bottom of my layout.vue:
```
<script>
  export default {
    middleware: 'authenticated-admin'
  }
</script>
```

This middleware checks if a user is authenticated or not.


In the index.vue of each folders I added:
  export default {
    middleware: 'auth-admin'  //or auth-user, auth-customer
  }


This seems to work for now, fingers crossed.

**Comment by sudhir600** [comment]
Hi @FreddyCrugger 
@awronski  answer is more suitable. i did almost same.  when user logged 1st time, i am getting permission  as json `(1:[1,2,3,4], 2[1,3]` like this. here key 1 and 2 is page id and 1, 2, 3, 4 is action id such as 1 = view, 2 = edit, 4= delete.
upon getting formula, save this in store (or in LS if you want), 
When any route change, check this with formula and then either allow to open or 401.

Although, front-end user can change this easily, so there is one more check at server label. Node server check what request is being called and what is the permission. if matched then user will get info.

However, it's still complex and repeated things. hope something will add in nuxt in future.

**Comment by KamranMaqbool** [comment]
any new update about this Question ?

**Comment by Arkadipta1998** [comment]
If anyones still interested... You can use the vue-simple-acl library. It's amazing and works well Nuxt 3 as well
